src/Base/Rotation.cpp

```diff
@@ -712,3 +713,271 @@ bool Rotation::isNull() const
             this->quat[2] == 0.0 &&
             this->quat[3] == 0.0);
 }
+
+//=======================================================================
+// The following code is borrowed from OCCT gp/gp_Quaternion.cxx

```

Comment:
That's not LGPL'ed code -- does their license allow this use?

Reply:
Really? But the license comment at the beginning of gp_Quaternion.cxx says its under LGPL. It also mentions that the implementation is based on other samples and posts in public.

Reply:
You are right, I was looking at an [old version](http://git.dev.opencascade.org/gitweb/?p=occt.git;a=blob;f=src/gp/gp_Quaternion..cxx;h=cb9ab504ca14cac70c8c77bb9e0ec35e60f14482;hb=f6f03db9d08051ceb1741ba8a18adfd172d98a91), they did re-license it when it got rewritten a few years ago. Thanks for checking.

Reply:
No problem. BTW, I'll respond to your other code comment later. You know it takes time to switch to the branch to modify and compile and all. I am currently busy with other stuff.